src/main/java/frc/robot/subsystems/arm/commands/SetArmsPosition.java

```diff
@@ -1,28 +1,37 @@
 package frc.robot.subsystems.arm.commands;
 
+import edu.wpi.first.math.MathUtil;
 import edu.wpi.first.math.geometry.Translation2d;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.arm.Arm;
-import         edu.wpi.first.math.MathUtil;
 import frc.robot.subsystems.arm.ArmConstants;
 
+import java.util.function.Function;
+import java.util.function.Supplier;
+
 public class SetArmsPosition extends CommandBase {
     private final Arm arm;
-    private final Translation2d position;
+    private final Supplier<Translation2d> position;
 
     public SetArmsPosition(Translation2d position) {
+        this(() -> position);
+    }
+
+    public SetArmsPosition(Supplier<Translation2d> position) {
         this.arm = Arm.getInstance();
         this.position = position;
         addRequirements(arm);
     }
 
     @Override
     public void initialize() {
-        arm.setEndPosition(position);
+        arm.setEndPosition(position.get());
     }
 
     @Override
     public boolean isFinished() {
-       return MathUtil.applyDeadband(position.getX() - arm.getEndPosition().getX(), ArmConstants.SETPOINT_DEADBAND) == 0 && MathUtil.applyDeadband(position.getY() - arm.getEndPosition().getY(), ArmConstants.SETPOINT_DEADBAND) == 0;
+        var position = this.position.get();
+        return MathUtil.applyDeadband(position.getX() - arm.getEndPosition().getX(), ArmConstants.SETPOINT_DEADBAND) == 0 &&
+                MathUtil.applyDeadband(position.getY() - arm.getEndPosition().getY(), ArmConstants.SETPOINT_DEADBAND) == 0;
```

Comment:
A slightly shorter solution:
```suggestion
        var difference = arm.getEndPosition().minus(position);
        return MathUtil.applyDeadband(difference.getX(), ArmConstants.SETPOINT_DEADBAND) == 0 &&
                MathUtil.applyDeadband(difference.getY(), ArmConstants.SETPOINT_DEADBAND) == 0;
```
Also, a deadband is not really the solution here -- when you apply a deadband on a joystick, you really want to change the output. Here, you just want to see if the values are closer to each other. So you can simply do the following:
```suggestion
        return Math.abs(difference.getX()) < ArmConstants.SETPOINT_DEADBAND && 
                Math.abs(difference.getY()) < ArmConstants.SETPOINT_DEADBAND;
```